### PR TITLE
minor fix: clean data type for negative operation

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

minor fix for negative operation in
1. negative physical expr
2. negative for scalar value

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/physical-expr/src/expressions/negative.rs [hunk 1]
```diff
@@ -118,13 +118,11 @@ pub fn negative(
     arg: Arc<dyn PhysicalExpr>,
     input_schema: &Schema,
 ) -> Result<Arc<dyn PhysicalExpr>> {
+    // TODO: the data type checker should be done in the logical plan phase
```
Comment (liukun4515) [inline datafusion/physical-expr/src/expressions/negative.rs:121]: If datafusion dose this in the preview stage or the logical plan generation stage, I think we don't need to check the child/input data type for the physical expr generation.

cc @alamb 
Is there user case that user generate the physical plan or physical expr directly without the logical plan or logical stage?

File: datafusion/common/src/scalar.rs [hunk 2]
```diff
@@ -710,24 +710,26 @@ impl ScalarValue {
...
+            ScalarValue::Int8(Some(v)) => Ok(ScalarValue::Int8(Some(-v))),
+            ScalarValue::Int16(Some(v)) => Ok(ScalarValue::Int16(Some(-v))),
+            ScalarValue::Int32(Some(v)) => Ok(ScalarValue::Int32(Some(-v))),
+            ScalarValue::Int64(Some(v)) => Ok(ScalarValue::Int64(Some(-v))),
             ScalarValue::Decimal128(Some(v), precision, scale) => {
-                ScalarValue::Decimal128(Some(-v), *precision, *scale)
+                Ok(ScalarValue::Decimal128(Some(-v), *precision, *scale))
             }
-            _ => panic!("Cannot run arithmetic negate on scalar value: {:?}", self),
+            value => Err(DataFusionError::Internal(format!(
```
Comment (liukun4515) [inline datafusion/common/src/scalar.rs:729]: replace the panic with the error
cc @andygrove

File: datafusion/physical-expr/src/expressions/negative.rs (hunk 1 above)
Comment (alamb) [reply datafusion/physical-expr/src/expressions/negative.rs:121]: I am not sure if anyone creates physical Exprs directly without a logical_plan Expr. 

However, I agree with this approach to throw an error if the physical types don't line up -- automatic coercion should be done at the logical level

**Review by alamb — APPROVED** [review]
LGTM -- thanks @liukun4515

File: datafusion/physical-expr/src/expressions/negative.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/physical-expr/src/expressions/negative.rs:121]: > I am not sure if anyone creates physical Exprs directly without a logical_plan Expr.
> 
> However, I agree with this approach to throw an error if the physical types don't line up -- automatic coercion should be done at the logical level

Got it. 
If the phy expr can be created directly without the type coercion or type checker in the logical phase, we need this checker in the creation of physical expr.

**Review by andygrove — APPROVED** [review]

File: datafusion/physical-expr/src/expressions/negative.rs (hunk 1 above)
Comment (alamb) [reply datafusion/physical-expr/src/expressions/negative.rs]: > If the phy expr can be created directly without the type coercion or type checker in the logical phase, we need this checker in the creation of physical expr.

That is a good point. 

Maybe the larger question is if we are ok requiring that the logical optimizer passes are run on `LogicalPlan`s and `Expr` to successfully create `ExecutionPlan`s and `PhysicalExpr`s, 

Now that I type that it seems there is value in supporting creating `Exprs` and then PhysicalExprs directly (as part of implementing extensions, for example) 🤔 So maybe we should just ensure it is possible to call the coercion logic directly on a `Expr`

**Pull request merged**